implementation/exporters/src/main/java/io/smallrye/opentelemetry/implementation/exporters/AbstractVertxExporterProvider.java

```diff
@@ -57,14 +62,28 @@ protected HttpExporter<T> createHttpExporter(ConfigProperties config, String htt
                 false);//TODO: this will be enhanced in the future
     }
 
+    /**
+     * If the CDI qualifier is specified in the config, it tries to get it from CDI, and if CDI does not provide such
+     * an instance on the specified qualifier, it leads to IllegalStateException.
+     * If the CDI qualifier is not specified in the config, it creates a new Vertx instance.
+     */
+    private Vertx getVertx(ConfigProperties config) {
+        String cdiQualifier = config.getString(OTEL_EXPORTER_VERTX_CDI_QUALIFIER);
+        if (cdiQualifier != null && !cdiQualifier.isEmpty()) {
+            return (Vertx) CDI.current().select(Identifier.Literal.of(cdiQualifier)).get();
```

Comment:
What happens if there is no Vertx bean defined? Say someone sets that identifier property (or it comes in via some dependency), but no actual bean is defined in the system for some strange reason. This look up will fail, correct? Should the system fail to start, or should it fallback to the default (possibly with a WARN message)?

Reply:
Yes, the look up will fail in that case. I am not sure if the system will fail to start, basically it will fail to `createExporter` I guess.

I think falling back to the default would be a better option. I think you mean `Vertx.vertx()` by `the default`, right ?

Reply:
That's what I had in mind, yes. I think NOT failing is probably the best option. It's certainly possible that the presence of that property with the absence of a bean is a bug, but it's certainly also possible that the entry is there unintentionally and not easily removed (e.g., third party jar), so I think a big, ugly warning and a reasonable fallback is probably the right choice.